aten/src/ATen/native/quantized/cpu/qsort.cpp

```diff
@@ -0,0 +1,51 @@
+#include <ATen/ATen.h>
+#include <ATen/core/op_registration/op_registration.h>
+#include <ATen/cpu/vec256/vec256.h>
+#include <ATen/native/SortingUtils.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cpu/Loops.h>
+#include <ATen/quantized/Quantizer.h>
+#include <ATen/native/quantized/cpu/quantized_ops.h>
+
+#include <algorithm>
+
+namespace at {
+namespace native {
+
+std::tuple<Tensor&, Tensor&> quantized_topk_out_cpu(
+    Tensor& values,
+    Tensor& indices,
+    const Tensor& self,
+    int64_t k,
+    int64_t dim_,
+    bool largest,
+    bool sorted) {
+  int64_t dim = maybe_wrap_dim(dim_, self.dim(), /*wrap_scalar=*/true);
+  TORCH_CHECK(
+      k >= 0 && k <= (self.dim() > 0 ? self.size(dim) : 1),
+      "selected index k out of range");
+  _allocate_or_resize_output_with_indices(values, indices, self, dim_, k);
+
+  qtopk_stub(kCPU, values, indices, self, k, dim, largest, sorted);
+
+  return std::forward_as_tuple(values, indices);
+}
+
+std::tuple<Tensor, Tensor> quantized_topk_cpu(
+    const Tensor& self,
+    int64_t k,
+    int64_t dim,
+    bool largest,
+    bool sorted) {
+  Tensor values = at::_empty_affine_quantized(
+    {0},
+    self.options(),
+    self.q_scale(),
```

Comment:
this will assert that quantizer is uniform, but it might be nice to add an explicit TORCH_CHECK for it